frame/bounties/src/lib.rs

```diff
@@ -715,6 +1441,102 @@ impl<T: Config> Module<T> {
 		Ok(())
 	}
 
+	fn ensure_bounty_active(
+		bounty_id: BountyIndex,
+	) -> Result<(T::AccountId, T::BlockNumber), DispatchError> {
+		let bounty = Self::bounties(&bounty_id).ok_or(Error::<T>::InvalidIndex)?;
+		if let BountyStatus::Active { curator, update_due } = bounty.status {
+			Ok((curator, update_due))
+		} else {
+			Err(Error::<T>::UnexpectedStatus.into())
+		}
+	}
+
+	fn create_subbounty(
+		bounty_id: BountyIndex,
+		subbounty_id: BountyIndex,
+		description: Vec<u8>,
+	) {
+
+		let subbounty = SubBounty {
+			fee: 0u32.into(),
+			curator_deposit: 0u32.into(),
+			status: SubBountyStatus::Added,
+		};
+
+		SubBounties::<T>::insert(bounty_id, subbounty_id, &subbounty);
+		BountyDescriptions::insert(subbounty_id, description);
+		Self::deposit_event(RawEvent::SubBountyAdded(bounty_id, subbounty_id));
+	}
+
+	fn impl_close_subbounty(
+		bounty_id: BountyIndex,
+		subbounty_id: BountyIndex,
+	) -> DispatchResult {
+		SubBounties::<T>::try_mutate_exists(
+			bounty_id,
+			subbounty_id,
+			|maybe_subbounty| -> DispatchResult {
+
+				let subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				match &subbounty.status {
+					SubBountyStatus::Added |
+					SubBountyStatus::SubCuratorProposed { .. } => {
+						// Nothing extra to do besides the removal of the bounty below.
+					},
+					SubBountyStatus::Active { subcurator } => {
+						// Cancelled by council, refund deposit of the working curator.
+						let _ = T::Currency::unreserve(subcurator, subbounty.curator_deposit);
+						// Then execute removal of the bounty below.
+					},
+					SubBountyStatus::PendingPayout { .. } => {
+						// Bounty is already pending payout. If council wants to cancel
+						// this bounty, it should mean the curator was acting maliciously.
+						// So the council should first unassign the curator, slashing their
+						// deposit.
+						return Err(Error::<T>::PendingPayout.into())
+					},
+				}
+
+				// Update the master curator fee &
+				// Reduce the active subbounty count.
+				Bounties::<T>::mutate_exists(
+					bounty_id,
+					|maybe_bounty| {
+						if let Some(bounty) = maybe_bounty.as_mut() {
+							bounty.fee = bounty
+								.fee
+								.saturating_add(subbounty.fee);
+							bounty.active_subbounty_count -= 1;
```

Comment:
use saturating_sub

Reply:
Here its subbounty cancel context, subbounty fee deducted during propose_subcurator() is reverted 
back to parent bounty.